File: Palindrom number/sol.java
class Solution {
    public boolean isPalindrome(int x) {
      
      if(x < 0) {
        return false; 
      }
      
      if(x <= 9) {
        return true; 
      }
      
      int num = x;
      int rev = 0, rem = 0; 
      
      while(num != 0) {
        rem = num % 10; 
        rev = rev * 10 + rem; 
        num /= 10; 
      }
      
      if(rev == x){
        return true;
      } else {
        return false; 
      }
    }
  }

File: Palindrom number/sol.py
'''
Time complexity : O(d) - d is the number of digits in the number 
'''
class Solution:
    def isPalindrome(self, x):
        '''
        :type x: int
        :rtype: bool
        '''
        if(x < 0):
            return False 
            
        if(x <= 9):
            return True 
            
        rev = 0
        num = x 
        while(num != 0): 
            rem = num % 10 
            rev = rev * 10 + rem 
            num = num // 10 
            
        if(rev == x):
            return True
        else: 
            return False 
